web_advanced_search/static/src/js/DomainSelector.esm.js

```diff
@@ -0,0 +1,43 @@
+/** @odoo-module **/
+
+import {patch} from "@web/core/utils/patch";
+import {_t} from "web.core";
+import Domain from "web.Domain";
+import DomainSelector from "web.DomainSelector";
+
+/**
+ * The redraw in the Debug Field does not trigger correctly
+ * so we overwrite it with the v14 Version
+ *
+ */
+patch(DomainSelector.prototype, "web.DomainSelector", {
+    /**
+     * @override
+     */
+    _onDebugInputChange: function (e) {
+        this._super(...arguments);
+
+        const rawDomain = e.currentTarget.value;
+        try {
+            Domain.prototype.stringToArray(rawDomain);
+        } catch (err) {
+            // If there is a syntax error, just ignore the change
+            this.displayNotification({
+                title: _t("Syntax error"),
+                message: _t("Domain not properly formed"),
+                type: "danger",
+            });
+            return;
+        }
+        if ($(".o_add_advanced_search").length) {
+            this._redraw(Domain.prototype.stringToArray(rawDomain)).then(
+                function () {
+                    this.trigger_up("domain_changed", {
+                        child: this,
+                        alreadyRedrawn: true,
+                    });
+                }.bind(this)
+            );
+        }
+    },
```

Comment:
I'd limit this override to the case we're on the advanced search dialog:
```suggestion
    /**
     * Our event is losing itself on our custom dialog. Just trigger the redraw
     * immediately
     * @override
     */
    _onDebugInputChange (e) {
        if (!$(".o_add_advanced_search").length) {
            return this._super(...arguments);
        }
        const rawDomain = e.currentTarget.value;
        try {
            Domain.prototype.stringToArray(rawDomain);
        } catch (err) { // If there is a syntax error, just ignore the change
            this.displayNotification({ title: _t("Syntax error"), message: _t("Domain not properly formed"), type: 'danger' });
            return;
        }
        this._redraw(Domain.prototype.stringToArray(rawDomain)).then(
            function () {
                this.trigger_up("domain_changed", {
                    child: this,
                    alreadyRedrawn: true,
                });
            }.bind(this)
        );
    },
```